ansible_base/authentication/authenticator_plugins/utils.py

```diff
@@ -43,7 +45,7 @@ def get_authenticator_plugin(authenticator_type: str):
     return AuthClass()
 
 
-def get_authenticator_urls(authenticator_type: str) -> list:
+def get_authenticator_urls(authenticator_type: str) -> Optional[ModuleType]:
```

Comment:
I think something is actually out of whack here....

If you look at `ansible_base.authentication.authenticator_plugins.saml` it defines urls as:
```
urls = [
    # SAML Metadata
    re_path(r'authenticators/(?P<pk>[0-9]+)/metadata/$', SAMLMetadataView.as_view(), name='authenticator-metadata'),
]
```

Based on the name of the function (and old type hint) I think the intention of this function is to return a list[URLPattern] but its returning the module and then, in the only place we call it (django-ansible-base/ansible_base/authentication/urls.py) we are doing:
```
for plugin_name in get_authenticator_plugins():
    plugin_urls = getattr(get_authenticator_urls(plugin_name), 'urls', None)
```

It seems like this function should be extracting the 'urls' filed from the module it gets (as `urls` in this function) and then returning that or `[]`.

Reply:
implemented